tests/unit/single_table/privacy/test_disclosure_protection.py

```diff
@@ -0,0 +1,254 @@
+            'col3': [np.nan, None, pd.NA],
+        })
+        synthetic_data = pd.DataFrame({
+            'col1': ['A', 'B', '__NULL_VALUE__'],
+            'col2': ['D', np.nan, 'A'],
+            'col3': [np.nan, None, pd.NA],
+        })
+        column_names = ['col1', 'col2']
+
+        # Run
+        null_category_map = DisclosureProtection._get_null_categories(
+            real_data, synthetic_data, column_names
+        )
+
+        # Assert
+        assert null_category_map == {'col1': '__NULL_VALUE___', 'col2': '__NULL_VALUE___'}
+
+    def test__discreteize_column(self):
+        """Test discretizing a continous column"""
+        # Setup
+        real_column = pd.Series([0, 2, 6, 8, 10])
+        synthetic_column = pd.Series([-10, 1, 3, 5, 7, 9, 20])
+
+        # Run
+        binned_real, binned_synthetic = DisclosureProtection._discretize_column(
+            real_column, synthetic_column, 5
+        )
+
+        # Assert
+        expected_real = pd.Series(pd.Categorical(['0', '0', '2', '3', '4']))
+        np.testing.assert_array_equal(binned_real, expected_real)
+        expected_synthetic = pd.Series(pd.Categorical(['0', '0', '1', '2', '3', '4', '4']))
+        np.testing.assert_array_equal(binned_synthetic, expected_synthetic)
+
+    def test__compute_baseline(self):
+        """Test computing the baseline score for random data."""
+        # Setup
+        real_data = pd.DataFrame({
+            'col1': ['A', 'A', 'A', 'A', 'A'],
+            'col2': ['A', 'B', 'A', 'B', 'A'],
+            'col3': range(5),
+        })
+        sensitive_column_names = ['col1', 'col2']
+
+        # Run
+        baseline_score = DisclosureProtection._compute_baseline(real_data, sensitive_column_names)
+
+        # Assert
+        assert baseline_score == 0.5
+
+    @patch('sdmetrics.single_table.privacy.disclosure_protection.CAP_METHODS')
+    def test_compute_breakdown(self, CAPMethodsMock):
+        """Test the ``compute_breakdown`` method."""
+        # Setup
+        real_data = pd.DataFrame({
+            'col1': np.random.choice(['A', 'B', 'C', 'D'], size=10),
+            'col2': ['X', 'Y', 'Z', None, np.nan, 'X', 'Y', 'Z', 'X', 'X'],
+            'col3': range(10),
+        })
+        synthetic_data = pd.DataFrame({
+            'col1': np.random.choice(['A', 'B', 'C', 'D'], size=10),
+            'col2': np.random.choice(['X', 'Y', 'Z', np.nan, None], size=10),
+            'col3': range(-2, 8),
+        })
+        CAPMock = Mock()
+        CAPMock.compute.return_value = 0.9
+        CAPMethodsMock.keys.return_value = ['CAP', 'ZERO_CAP', 'GENERALIZED_CAP']
+        CAPMethodsMock.get.return_value = CAPMock
+
+        # Run
+        score_breakdown = DisclosureProtection.compute_breakdown(
+            real_data=real_data,
+            synthetic_data=synthetic_data,
+            known_column_names=['col1'],
+            sensitive_column_names=['col2', 'col3'],
+            continuous_column_names=['col3'],
```

Comment:
Should we verify that we support all dtypes with `continuous_column_names`? Float64, Float32, Int64, Int32 are some additional ones to look into (beyond float64 and int64 (numpy dtypes)).